lib/app/server.interfaces.ts

```diff
@@ -0,0 +1,80 @@
+// Copyright (c) 2025, Compiler Explorer Authors
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+//     * Redistributions of source code must retain the above copyright notice,
+//       this list of conditions and the following disclaimer.
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+// LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+// SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+// CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+// ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+// POSSIBILITY OF SUCH DAMAGE.
+
+import type {Express, Request, Response, Router} from 'express';
+import type {GoldenLayoutRootStruct} from '../clientstate-normalizer.js';
+import type {ShortLinkMetaData} from '../handlers/handler.interfaces.js';
+import type {PropertyGetter} from '../properties.interfaces.js';
+
+export interface ServerOptions {
+    staticPath: string;
+    staticMaxAgeSecs: number;
+    staticUrl?: string;
+    staticRoot: string;
+    httpRoot: string;
+    sentrySlowRequestMs: number;
+    distPath: string;
+    extraBodyClass: string;
+    maxUploadSize: string;
+}
+
+export interface PugOptions {
+    extraBodyClass: string;
+    httpRoot: string;
+    staticRoot: string;
+    storageSolution: string;
+    optionsHash: string;
+    compilerExplorerOptions: string;
+}
+
+export interface RenderConfig extends PugOptions {
+    embedded: boolean;
+    mobileViewer: boolean;
+    readOnly?: boolean;
+    config?: GoldenLayoutRootStruct;
+    metadata?: ShortLinkMetaData;
+    storedStateId?: string | false;
+    [key: string]: any;
+}
+
+export type PugRequireHandler = (path: string) => string;
```

Comment:
Could you add a TODO or something here (feel free to put my name on it) that this thing will be removed in the future and should ideally not be used?

Reply:
Will do